ENIGMAsystem/SHELL/Universal_System/fileio.cpp

```diff
@@ -164,14 +164,15 @@ double file_text_read_real(int fileid) { // Reads a real value from the file and
   return r1;
 }
 
-void file_text_readln(int fileid) // Skips the rest of the line in the file and starts at the start of the next line.
+std::string file_text_readln(int fileid) // Skips the rest of the line in the file and starts at the start of the next line.
 {
-  if (feof(enigma::files[fileid].f))
-    enigma::files[fileid].eof = true;
+  enigma::openFile &mf = enigma::files[fileid];
+  if (feof(mf.f))
+    mf.eof = true;
   string ret;
```

Comment:
I'm not sure why this is called "ret" if we were never returning it. Perhaps at some point, someone *was* returning it, then realized it wasn't right and made it `void` instead.

Reply:
Since I've already made the diff big for this, I guess I will rename it as I did in #1141. I will call the value returned `ret` and the one cached for later `next` as in next line, since that's what it is.

Reply:
Ok: 5866d2997afd900c0aaf1dfb89ef90abaeffcbf8

I used next for the next line and last for the line we are returning.